cadence/contracts/bridge/FlowEVMBridge.cdc

```diff
@@ -74,14 +109,37 @@ contract FlowEVMBridge : IFlowEVMNFTBridge {
         let feeVault <-feeProvider.withdraw(amount: FlowEVMBridgeConfig.onboardFee) as! @FlowToken.Vault
         FlowEVMBridgeUtils.deposit(<-feeVault)
         // Deploy an EVM defining contract via the FlowBridgeFactory.sol contract
-        let erc721Address = self.deployEVMContract(forAssetType: type)
+        // let evmContractAddress = self.deployEVMContract(forAssetType: type)
+        let onboardingValues = self.deployEVMContract(forAssetType: type)
         // Initialize bridge escrow for the asset
-        FlowEVMBridgeNFTEscrow.initializeEscrow(forType: type, erc721Address: erc721Address)
+        if type.isSubtype(of: Type<@{NonFungibleToken.NFT}>()) {
+            FlowEVMBridgeNFTEscrow.initializeEscrow(
+                forType: type,
+                name: onboardingValues.name,
+                symbol: onboardingValues.symbol,
+                erc721Address: onboardingValues.evmContractAddress
+            )
+        } else {
```

Comment:
Might want to make the check explicit for fungible tokens and panic if neither NFT or FT

Reply:
Since that check is done in several places, I include it in `typeRequiresOnboarding()` which is validated in the pre-condition. Beyond the pre-check, we know the type is either an NFT or FT. By the point of the `if` block above, we know it's either NFT or FT, and by the `else` block it must be FT. Do you think I should organize these checks differently?

Reply:
Makes sense.
I personally like very explicit checks because sometimes assumptions break with future code updates. Not confident it applies here tho

Reply:
I can see how it would be more resilient to add the explicit checks here. Doesn't hurt to add them so I'll include an else if + else -> panic